### DISCUSSION TITLE
Can Kong ingress control traffic between two k8s deployment which relate to one k8s service like istio traffic-shifting?

### DISCUSSION THREAD

**Discussion by gung-rgb** [Kong/kong#8651] [Help]
[istio traffic-shifting](https://istio.io/latest/docs/tasks/traffic-management/traffic-shifting/) can create two deployment `reviews-v1`、`reviews-v2`, relate to one k8s service: `reviews`. And then, we can config the `VirtualService` to control some traffic send to `deployment-v1`, others send to v2 by conditions, like by header、by querystring, or just adjust weight between the two deployment.

For kong ingress, how to achieve these goal? 

for my view,  in kong data plane(kong proxy), only one upstream relate to one k8s service,  also we can not distinguish targets between v1 and v2 by that one upstream, so we can not achieve it by custom plugins. or something i miss?

### Proposed Solution

1.  Create two upstream relate to two deployment.
2. add tag to distinguish deployment from v1 to v2. But kong balance know nothing about tag, so we can not use kong balance to control traffic.  Or Maybe our plugin subscribe the target change event and create our upstream by plugin code?

**Comment by hbagdi** [comment]
cc @Kong/team-k8s
